lib/ads/interstitial_ad_manager.js

```diff
@@ -203,48 +203,63 @@ shaka.ads.InterstitialAdManager = class {
 
 
   /**
-   * @param {shaka.extern.Interstitial} interstitial
+   * @param {shaka.extern.HLSInterstitial} hlsInterstitial
    */
-  async addMetadata(interstitial) {
+  async addMetadata(hlsInterstitial) {
     if (this.basePlayer_.getLoadMode() == shaka.Player.LoadMode.SRC_EQUALS &&
         this.usingBaseVideo_) {
       shaka.log.alwaysWarn(
-          'Unsupported interstitial when using single media element',
-          interstitial);
+          'Unsupported HLS interstitial when using single media element',
+          hlsInterstitial);
       return;
     }
     this.updatePlayerConfig_();
+    const adInterstitials = await this.getInterstitialsInfo_(hlsInterstitial);
+    if (adInterstitials.length) {
+      this.addInterstitials(adInterstitials);
+    } else {
+      shaka.log.alwaysWarn('Unsupported HLS interstitial', hlsInterstitial);
+    }
+  }
+
+
+  /**
+   * @param {!Array.<shaka.extern.AdInterstitial>} interstitials
+   */
+  addInterstitials(interstitials) {
     let cuepointsChanged = false;
-    const interstitialsAd = await this.getInterstitialsInfo_(interstitial);
-    if (interstitialsAd.length) {
-      for (const interstitialAd of interstitialsAd) {
-        const interstitialId = JSON.stringify(interstitialAd);
-        if (this.interstitialIds_.has(interstitialId)) {
-          continue;
-        }
-        cuepointsChanged = true;
-        this.interstitialIds_.add(interstitialId);
-        this.interstitials_.add(interstitialAd);
-        let shouldPreload = false;
-        if (interstitial.pre && this.lastTime_ == null) {
-          shouldPreload = true;
-        } else if (interstitialAd.startTime == 0 && !interstitialAd.canJump) {
+    for (const interstitial of interstitials) {
+      if (this.basePlayer_.getLoadMode() == shaka.Player.LoadMode.SRC_EQUALS &&
```

Comment:
This check can probably be outside of the for loop, at the top of the method.

Reply:
Done!